src/service/sharding-table-service.js

```diff
@@ -158,25 +159,43 @@ class ShardingTableService {
         contentAssetStorageAddress,
         firstAssertionId,
         hashFunctionId,
+        option,
     ) {
         const kbSize = assertionSize < BYTES_IN_KILOBYTE ? BYTES_IN_KILOBYTE : assertionSize;
+        const r0 = await this.blockchainModuleManager.getR0(blockchainId);
```

Comment:
Would it be possible to get r0, r1 and r2 only when needed and / or to parallelize these calls ?